src/hotspot/share/memory/metaspace.cpp

```diff
@@ -586,11 +586,14 @@ ReservedSpace Metaspace::reserve_address_space_for_compressed_classes(size_t siz
       (char*) CompressedKlassPointers::reserve_address_space_for_compressed_classes(size, RandomizeClassSpaceLocation,
                                                                                     optimize_for_zero_base));
 
+#if !defined(AARCH64)
+  // On aarch64, this likely wouldn't satisfy the encoding constraints.
```

Comment:
Please don't add platform ifdefs here! We much work into getting rid of platform-dependent sections. Also, as stated above, I think this change is not useful.